test/e2e/node/security_context.go

```diff
@@ -128,6 +128,48 @@ var _ = SIGDescribe("Security Context [Feature:SecurityContext]", func() {
 		})
 	})
 
+	It("should support container.SecurityContext.RunAsNonRootGroup [Feature:RunAsGroup]", func() {
+		pod := scTestPod(false, false)
+		userID := int64(1001)
+		zeroGroupID := int64(0)
+		nonZeroGroupID := int64(2004)
+		trueVal := true
+		pod.Spec.SecurityContext.RunAsNonRootGroup = &trueVal
+		pod.Spec.SecurityContext.RunAsUser = &userID
+
+		// Zero RunAsGroup should fail
+		pod.Spec.SecurityContext.RunAsGroup = &zeroGroupID
+		pod.Spec.Containers[0].Command = []string{"sh", "-c", "id"}
+
+		client := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+		pod, err := client.Create(pod)
+		Expect(err).To(HaveOccurred())
+
+		// Zero FSGroup should fail
+		pod.Spec.SecurityContext.RunAsGroup = nil
+		pod.Spec.SecurityContext.FSGroup = &zeroGroupID
+		pod, err = client.Create(pod)
+		Expect(err).To(HaveOccurred())
+
+		// Zero SupplementalGroup should fail
```

Comment:
nit: make each of these a different test (`It(...)`), and grouped by `Context(...)`